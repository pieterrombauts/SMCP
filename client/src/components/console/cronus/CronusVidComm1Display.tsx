import React from 'react';
import styled from 'styled-components';
import { cronusVComm1Display } from '../customTypes'

interface CronusVC1DProps {
  className?: string;
  values: cronusVComm1Display;
};

const UCronusVidComm1Display: React.FC<CronusVC1DProps> = ( props ) => {
  return(
    <div className={props.className}>
      <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
        <image x={0} y={0} width={1300} height={990} href='/media/Cronus_4.png'></image>
        <rect x="450" y="375" rx="5" ry="5" width="435" height="335" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}}/>

        <text x="37%" y="45%" fill="white" fontSize="45" fontFamily="Verdana">Ku-Band Status</text>

        <text x="37%" y="50%" fill="white" fontSize="20" fontFamily="Verdana">Downlink 1 Status</text>
        <text x="55%" y="50%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">ACTIVE</text>

        <text x="37%" y="53%" fill="white" fontSize="20" fontFamily="Verdana">Downlink 2 Status</text>
        <text x="55%" y="53%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">ACTIVE</text>

        <text x="37%" y="56%" fill="white" fontSize="20" fontFamily="Verdana">Downlink 3 Status</text>
        <text x="55%" y="56%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">ACTIVE</text>

        <text x="37%" y="59%" fill="white" fontSize="20" fontFamily="Verdana">Downlink 4 Status</text>
        <text x="55%" y="59%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">ACTIVE</text>


        <text x="37%" y="64%" fill="white" fontSize="20" fontFamily="Verdana">Elevation Gimbal</text>
        <text x="58%" y="64%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">{props.values.vComms1.elevationGimbal}</text>
        <text x="63%" y="64%" fill="white" fontSize="20" fontFamily="Verdana">°</text>

        <text x="37%" y="67%" fill="white" fontSize="20" fontFamily="Verdana">Cross-Elevation Gimbal</text>
        <text x="58%" y="67%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">{props.values.vComms1.crossElevationGimbal}</text>
        <text x="63%" y="67%" fill="white" fontSize="20" fontFamily="Verdana">°</text>
      </svg>
    </div>
  )
};

const CronusVidComm1Display = styled(UCronusVidComm1Display)`
  svg {
    width: 100%;
    height: 100%;
  }
`;

export default CronusVidComm1Display;