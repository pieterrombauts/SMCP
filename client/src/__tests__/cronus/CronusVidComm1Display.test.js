import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import CronusVidComm1Display from '../../components/console/cronus/CronusVidComm1Display'
import { shallow, mount, render} from 'enzyme';

const defaultValues = {
  vComms1: {
    elevationGimbal: 48,
    crossElevationGimbal: 45
  }
}

describe("Cronus UHF communications display", () => {
  it('renders correctly', () => {
    const wrapper = mount(<CronusVidComm1Display values={defaultValues} />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})