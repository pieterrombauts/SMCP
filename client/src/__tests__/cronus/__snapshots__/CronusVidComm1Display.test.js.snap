// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cronus UHF communications display renders correctly 1`] = `
.c0 svg {
  width: 100%;
  height: 100%;
}

<Styled(UCronusVidComm1Display)
  values={
    Object {
      "vComms1": Object {
        "crossElevationGimbal": 45,
        "elevationGimbal": 48,
      },
    }
  }
>
  <UCronusVidComm1Display
    className="c0"
    values={
      Object {
        "vComms1": Object {
          "crossElevationGimbal": 45,
          "elevationGimbal": 48,
        },
      }
    }
  >
    <div
      className="c0"
    >
      <svg
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 1300 990"
      >
        <image
          height={990}
          href="/media/Cronus_4.png"
          width={1300}
          x={0}
          y={0}
        />
        <rect
          height="335"
          rx="5"
          ry="5"
          style={
            Object {
              "fill": "#36485b",
              "stroke": "#2a3a3f",
              "strokeWidth": "5",
            }
          }
          width="435"
          x="450"
          y="375"
        />
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="45"
          x="37%"
          y="45%"
        >
          Ku-Band Status
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="37%"
          y="50%"
        >
          Downlink 1 Status
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="55%"
          y="50%"
        >
          ACTIVE
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="37%"
          y="53%"
        >
          Downlink 2 Status
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="55%"
          y="53%"
        >
          ACTIVE
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="37%"
          y="56%"
        >
          Downlink 3 Status
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="55%"
          y="56%"
        >
          ACTIVE
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="37%"
          y="59%"
        >
          Downlink 4 Status
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="55%"
          y="59%"
        >
          ACTIVE
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="37%"
          y="64%"
        >
          Elevation Gimbal
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="58%"
          y="64%"
        >
          48
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="63%"
          y="64%"
        >
          °
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="37%"
          y="67%"
        >
          Cross-Elevation Gimbal
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="58%"
          y="67%"
        >
          45
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="63%"
          y="67%"
        >
          °
        </text>
      </svg>
    </div>
  </UCronusVidComm1Display>
</Styled(UCronusVidComm1Display)>
`;
